crs_support/launch/environment_launch.launch.py

```diff
@@ -7,25 +8,49 @@
 import launch
 import launch_ros.actions
 from launch import some_substitutions_type
+from rosidl_generator_cpp import default_value_from_type
 
 def generate_launch_description():
 
     if not "tesseract_collision" in os.environ["AMENT_PREFIX_PATH"]:
         # workaround for pluginlib ClassLoader bug: manually add tesseract_collision to the AMENT_PREFIX_PATH env variable
         head, tail = os.path.split(get_package_prefix('crs_support'))
         path = os.path.join(head, 'tesseract_collision')
-        os.environ["AMENT_PREFIX_PATH"] += os.pathsep + path
-
+        os.environ["AMENT_PREFIX_PATH"] += os.pathsep + path  
+    
+    
+    xacro = os.path.join(get_package_share_directory('crs_support'), 'urdf', 'crs.urdf.xacro')
     urdf = os.path.join(get_package_share_directory('crs_support'), 'urdf', 'crs.urdf')
+    urdf_preview = os.path.join(get_package_share_directory('crs_support'), 'urdf', 'crs_preview.urdf')
     srdf = os.path.join(get_package_share_directory('crs_support'), 'urdf', 'ur10e_robot.srdf')
     gzworld = os.path.join(get_package_share_directory('crs_support'), 'worlds', 'crs.world')
     
-    # kill any lingering gazebo instances first
+    # create urdfs from xacro file
+    cmd2 = 'xacro %s prefix:=preview/ > %s'%(xacro, urdf_preview)
+    cmd3 = 'xacro %s > %s'%(xacro, urdf)        
+    cmd4 = 'killall -9 gazebo & killall -9 gzserver & killall -9 gzclient'
+    cmd_dict = {cmd2 : True, cmd3: True, cmd4 : False}
+    
+    # check if soft link exists
+    gazebo_model_path = os.path.join(os.environ['HOME'],'.gazebo', 'models', 'crs_support')
+    crs_model_path = get_package_share_directory('crs_support')
+    cmd1 = 'ln -s %s %s'%(crs_model_path, gazebo_model_path)
+    if not os.path.exists(gazebo_model_path):
+        cmd_dict[cmd1] = True        
+    
+    required_cmd = False
     try:    
-        cmd = 'killall -9 gazebo & killall -9 gzserver & killall -9 gzclient'
-        process = subprocess.run(cmd , shell=True, check=True, stdout=subprocess.PIPE, universal_newlines=True)
+
+        for cmd, req_ in cmd_dict.items():       
```

Comment:
The try-except should be inside the for loop. The way it is currently set up is if the cmd4 fails, which it often does if gazebo isn't running, then it won't even try cmd1.